src/brod_utils.erl

```diff
@@ -172,8 +148,49 @@ assert_topic(Topic) ->
   ok_when(is_binary(Topic) andalso size(Topic) > 0,
           {bad_topic, Topic}).
 
+%% @doc Map message to brod's format.
+%% incomplete message indicator is kept when the only one message is incomplete.
+%% Messages having offset earlier than the requested offset are discarded.
+%% this might happen for compressed message sets
+%% @end
+-spec map_messages(offset(), [ {?incomplete_message, non_neg_integer()}
+                             | kpro_Message()
+                             ]) ->
+       {?incomplete_message, non_neg_integer()} | [#kafka_message{}].
+map_messages(BeginOffset, Messages) when is_binary(Messages) ->
+  map_messages(BeginOffset, kpro:decode_message_set(Messages));
+map_messages(_BeginOffset, [{?incomplete_message, Size}]) ->
+  {?incomplete_message, Size};
+map_messages(BeginOffset, Messages) when is_list(Messages) ->
+  [kafka_message(M) || M <- Messages,
+   is_record(M, kpro_Message) andalso
+   M#kpro_Message.offset >= BeginOffset].
+
+
 %%%_* Internal Functions =======================================================
 
+%% @private Convert a `kpro_Message' to a `kafka_message'.
+-spec kafka_message(#kpro_Message{})     -> #kafka_message{};
+                   (?incomplete_message) -> ?incomplete_message.
+kafka_message(#kpro_Message{ offset     = Offset
```

Comment:
no, field `messageSize` from kpro record is not included in `#kafka_message{}`.
and there might be more fields added in the next kafka protocol version which are useless for consumers.
Or perhaps in 3.0, we can simply expose kpro records.